optashift-employee-rostering-shared/src/main/java/org/optaplanner/openshift/employeerostering/shared/roster/RosterState.java

```diff
@@ -63,15 +65,30 @@ public boolean isDraft(OffsetDateTime dateTime) {
     }
 
     @JsonIgnore
-    public boolean isHistoric(Shift shift) {
-        return isHistoric(shift.getStartDateTime());
+    public boolean isPublished(OffsetDateTime dateTime) {
+        return !isHistoric(dateTime) && !isDraft(dateTime);
```

Comment:
isHistoric must be before isDraft which must be before isPublished.
Please revert these changes.

Reply:
ah, I get what happened. Just move isPublished(OffsetDatetime) before or after the other isPublished(LocalDateTime) method